src/hosted_db/paths.rs

```diff
@@ -21,35 +25,75 @@ pub fn database_parent_path(data_path: &str, create_path: bool) -> Result<PathBu
     Ok(fs::canonicalize(path)?)
 }
 
-pub fn database_path(
+/// Returns a path for a new database directory for storing
+/// `{entity_slug}/{database_slug}`. The format for this path is
+/// `{data_path}/{entity_slug}/{database_slug}/{time_sortable_uuid}/`.
+pub fn new_database_path(
     entity_slug: &str,
     database_slug: &str,
     data_path: &str,
-    create_database: bool,
 ) -> Result<PathBuf, AybError> {
+    let uuid = Uuid::new_v7(Timestamp::now(ContextV7::new()));
```

Comment:
I'm interested, why do you use UUIDv7? :)

Reply:
Great question! Sorry for the delay in replying---I created the PR and went on vacation :).

In practice, several versions of UUID would work in this specific situation, but here was my reasoning:
* The big downside of v7/time-sortable UUIDs is that they might leak sensitive details, namely the quantity/frequency of UUIDs that are created. If you're trying to hide, for example, how many customers you have, UUIDv7 will leak details similar to how serial integer primary keys do. We don't externally expose the UUIDs of these directories, and even if we did leak them, I couldn't think of a reason why the creation timestamp of a database's folder is sensitive information.
* One very weak benefit of v7 is, because it's time-sortable, if you ever end up in a corrupted state and need to manually dig through the directory structure, being able to determine which folder was created before/after other folders might help you recover the most recent snapshot/version of a database. (This information would also be available in filesystem and S3 metadata, so it's only a minor benefit.)
* Of all of the time-sortable UUID versions, v7 has the benefit that, when stored as bytes in PostgreSQL (or likely any B-Tree index), there is better temporal locality if you're inserting roughly in time order, which has performance and space benefits [1, 2]. We're not storing these particular UUID instances in a database, but if we're using any time-sortable UUID in the project and v7 outperforms the others when stored in a database, we may as well use that same version of UUID in other parts of the project all other things being equal.

[1] https://ardentperf.com/2024/02/03/uuid-benchmark-war/
[2] https://maciejwalkowiak.com/blog/postgres-uuid-primary-key/

Reply:
Thank you for the thorough response! I don't know much about UUIDs aside from v4, so this is really insightful :)